dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/kdoc/ResolveKDocLink.kt

```diff
@@ -70,15 +82,17 @@ private fun KaSession.createKDocLink(link: String, context: PsiElement? = null):
  *
  * @return [DRI] or null if the [link] is unresolved
  */
-internal fun KaSession.resolveKDocLinkToDRI(link: KDocLink): DRI? {
-    val linkedSymbol = resolveToSymbol(link)
-    return if (linkedSymbol == null) null
-    else getDRIFromSymbol(linkedSymbol)
+internal fun resolveKDocLinkToDRI(kDocLink: KDocLink): DRI? {
+    analyze(kDocLink) {
```

Comment:
As far as I understand, this will be called on each kdoc link resolution, am I right? Can it cause a degradation in performance?
Also, it's not clear for me, why we need an additional `analyze` call here. Could you please explain why having an already established session for the module is not enough?

Reply:
> Can it cause a degradation in performance?

It should not degrade performance since creating KaSessions is cacheable. 


> Could you please explain why having an already established session for the module is not enough?
 
 I left a comment in code